### PR TITLE
Logging rework

### PR DESCRIPTION
**Pull request by MPM1107**
Addresses #2373 and includes changes from #2231 

Had some issues figuring out how to write the tests, they probably need a closer look to check if everything is alright. Otherwise all functions of the code are tested and working.

TODO: Update documentation to address the changes

### CODE REVIEW

File: lib/util/settings.js [hunk 1]
```diff
@@ -42,7 +42,9 @@ const defaults = {
         output: 'json',
     },
     advanced: {
-        log_directory: path.join(data.getPath(), 'log', '%TIMESTAMP%'),
+        log_symlink: false,
+        log_output: ['console', 'file'],
```
Comment (Koenkk) [inline lib/util/settings.js]: Is it still possible to log to both console and file?

File: lib/util/yaml.js [hunk 2]
```diff
@@ -29,6 +29,9 @@ function readIfExists(file) {
 function writeIfChanged(file, content) {
     const before = readIfExists(file);
     if (!equals(before, content)) {
+        console.log(`Writing to yaml`);
+        console.log(before);
```
Comment (Koenkk) [inline lib/util/yaml.js]: I think this should be removed.

**Comment by Koenkk** [comment]
Can you provide some more context on what problems you try to address in this pr?

File: lib/util/settings.js (hunk 1 above)
Comment (MPM1107) [reply lib/util/settings.js]: The way the default is set enables logging for both console and file. log_directory is an array which includes "console", "file", both or an empty array. Including both "console" and "file" enables the current logging behaviour (console and file).

File: lib/util/yaml.js (hunk 2 above)
Comment (MPM1107) [reply lib/util/yaml.js]: I based everything I did one the other logging-related pull request (#2231 ), which had this in it. Looking at this again I would agree that this piece of code is not really necessary.

**Comment by MPM1107** [comment]
I found the current logging structure with multiple folders which each include multiple files a bit confusing, so using the daily-rotate-file transport simplifies logs into a single folder with separate files for every day, which also get cleaned up automatically after a max file count is reached.

In addition there is the option to enable a symlink to the current log. That way, if you want to read the most recent log file, you just open that symlink file instead of searching for the latest file manually. This also makes it easier to read the most recent log using software.

If you don't agree with the changes or have any comments I'm happy to change things around

**Comment by Koenkk** [comment]
The reason why we originally started logging per start date was to easily identify logs of different zigbee2mqtt runs. One directory contains now max 3 log files of a single zigbee2mqtt run. 

Currently it is already possible to log to a single directory by settings

```yaml
advanced:
  log_directory: data/log
```

Would this already cover your use case?

I however for sure like the other changes in this PR (symlink and configurable transports).

**Pull request closed without merging**